src/plot/br-heatmap.ts

```diff
@@ -42,60 +44,54 @@ export class BrHeatmap extends Plot {
         await this.legend.update();
     }
 
-    get mouseleaveEvent(): () => void {
-        const self = this;
-        return function(): void {
-            d3.select(this).style("stroke", "black");
-            self.tooltip.hide();
-        };
+    static eventWrapper<S extends SVGRectElement | SVGSVGElement, T extends (d: SquareInfo, node: S) => void | Promise<void>>(cb: T)
+    : (this: S, d: SquareInfo, i: number, n: S[]) => void {
+        return (d, i, n) =>
+            // https://stackoverflow.com/questions/27746304/how-to-check-if-an-object-is-a-promise/27760489#27760489
+            Promise.resolve((cb.bind(this) as T)(d, n[i])).then(() => {});
```

Comment:
The tooltip doesn't work after updating. Please revise it.

br-heatmap.ts:69 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'update')
    at Function.<anonymous> (br-heatmap.ts:69:28)
    at step (color-map-def.ts:44:68)
    at Object.next (color-map-def.ts:44:68)
    at color-map-def.ts:44:68
    at new Promise (<anonymous>)
    at __webpack_modules__../src/plot/br-heatmap.ts.__awaiter (color-map-def.ts:44:68)
    at __webpack_modules__../src/plot/br-heatmap.ts.BrHeatmap.onPointerMove (br-heatmap.ts:68:60)
    at SVGRectElement.<anonymous> (br-heatmap.ts:51:49)
    at SVGRectElement.<anonymous> (on.js:27:1)